vyper/codegen/module.py

```diff
@@ -91,6 +91,11 @@ def _is_internal(func_ast):
 def _is_payable(func_ast):
     return func_ast._metadata["type"].mutability == StateMutability.PAYABLE
 
+def _is_zero_selector_func(f):
+    abi_sig = f._metadata["signature"].abi_signature_for_kwargs([])
+    selector = utils.abi_method_id(abi_sig)
+    if str(selector)[0] == '0':
```

Comment:
probably want `if selector == b"\x00"*4` here, unless we decide we want the check to be for trailing zeroes. also above could use a sanity check that `len(selector) == 4`

Reply:
the selector is an int, once its been run through abi_method_id; therefore len(selector) == 4 will fail for the zero selector. I can run hex(selector) to get it in the form suggested, but the zero selector will still equal '0x0', not '0x00000000'. Perhaps it would be good to catch all selectors with trailing zeroes, as well?

Reply:
fixed with:
if selector.to_bytes(4, 'big') == b"\x00"*4: